Sources/JExtractSwiftLib/JNI/JNISwift2JavaGenerator+NativeTranslation.swift

```diff
@@ -917,6 +943,70 @@ extension JNISwift2JavaGenerator {
         printer.print("}")
 
         return printer.finalize()
+      
+      case .escapingClosureLowering(let parameters, let nativeResult, let closureName):
+        var printer = CodePrinter()
+
+        let methodSignature = MethodSignature(
+          resultType: nativeResult.javaType,
+          parameterTypes: parameters.flatMap {
+            $0.parameters.map { parameter in
+              guard case .concrete(let type) = parameter.type else {
+                fatalError("Closures do not support Java generics")
+              }
+              return type
+            }
+          }
+        )
+
+        printer.print("{")
+        printer.indent()
+        printer.print("let globalRef_\(closureName)$ = environment.interface.NewGlobalRef(environment, \(placeholder))!")
+        printer.print("let closureContext_\(closureName)$ = JNIClosureContext(globalRef: globalRef_\(closureName)$)")
+        printer.print("")
+        
+        let names = parameters.flatMap { $0.parameters.map(\.name) }
+        let closureParameters = !parameters.isEmpty ? "\(names.joined(separator: ", ")) in" : ""
+        printer.print("return { \(closureParameters)")
+        printer.indent()
+        
+        printer.print("guard let env$ = try? JavaVirtualMachine.shared().environment() else {")
+        printer.indent()
+        printer.print("fatalError(\"Failed to get JNI environment for escaping closure call\")")
+        printer.outdent()
+        printer.print("}")
+        printer.print("")
```

Comment:
Generally we do a big print with """ strings rather than this style when possible